src/bootstrap/dist.rs

```diff
@@ -45,6 +46,18 @@ fn missing_tool(tool_name: &str, skip: bool) {
     }
 }
 
+fn should_build_extended_tool(builder: &Builder<'_>, tool: &str) -> bool {
+    if !builder.config.extended {
+        return false;
+    }
+
+    if let Some(tools) = &builder.config.tools {
+        tools.is_empty() || tools.contains(tool)
```

Comment:
Can we refactor tools to be an `Option<Vec<...>>` or something to avoid `tools = []` to indicate all tools (which seems like not what the user wants)?

Reply:
I agree with your reading that `build.tools = []` should not build any tool instead of building all of them, I updated the code. By the way, that config option was already represented as `Option<HashSet<T>>`, I just implemented my check wrong.